### DISCUSSION TITLE
How to identify extension built-in assumptions?

### DISCUSSION THREAD

**Discussion by robertoaloi** [community/community#5110] [Codespaces] [answered]
Many VS Code extensions are not available in Codespaces. For example, here's the current status of Erlang and Rust extensions:

<img width="200" alt="Screenshot 2021-08-15 at 15 38 21" src="https://user-images.githubusercontent.com/91769/129481232-126e6682-cf70-439a-b5e7-f8881715c2e2.png"> <img width="200" alt="Screenshot 2021-08-15 at 15 38 07" src="https://user-images.githubusercontent.com/91769/129481243-4c1ab80d-3ee7-4942-8fab-04ccc570a850.png">

According [to the docs](https://code.visualstudio.com/docs/remote/codespaces#_why-is-an-extension-not-installable-in-the-browser), this is due to the extensions "having built-in assumptions or needing to run on the desktop".

As an extension developer, what's a good strategy to identify (and eventually eliminate) such assumptions?

**Comment by tanmayeekamath** [comment, marked as answer]
Hey @robertoaloi! Thanks for your question and feedback on this! 

The extensions that don't work in Codespaces aren't enabled for now, but there isn't a set list of "why" - it can be extension-specific (i.e. their own dependencies). I understand the "built in assumptions" phrase is a bit confusing, we will update that in the docs. 

That said we are also working on a doc that explains what needs to be done to run an extension in the web that will be out soon! You can find the work item in our [public plan here (extensions section)](https://github.com/microsoft/vscode/issues/130368). 

Hope this helps! :)
